csm/conf/configure.py

```diff
@@ -257,3 +260,67 @@ def get_component_list_from_features_endpoints():
         except Exception as e_:
             Log.error(f"Error in storing unsupported features: {e_}")
             raise CsmSetupError(f"Error in storing unsupported features: {e_}")
+
+    def _configure_csm_ldap_schema(self):
+        """
+        Configure openLdap for CORTX Users
+        """
+        Log.info("Openldap configuration started for Cortx users.")
+        _rootdnpassword = self._fetch_ldap_root_password()
+        if not _rootdnpassword:
+            raise CsmSetupError("Failed to fetch LDAP root user password.")
+
+        # Insert cortxuser schema
+        self._run_ldap_cmd(f'ldapadd -x -D cn=admin,cn=config -w {_rootdnpassword} -f {const.CORTXUSER_SCHEMA_LDIF} -H ldapi:///')
+
+        # Initialize dc=csm,dc=seagate,dc=com
+        self._run_ldap_cmd(f'ldapadd -x -D cn=admin,dc=seagate,dc=com -w {_rootdnpassword} -f {const.CORTXUSER_INIT_LDIF} -H ldapi:///')
+
+        # Setup necessary permissions
+        self._setup_ldap_permissions()
+
+        # Create Cortx Account
+        self._run_ldap_cmd(f'ldapadd -w {_rootdnpassword} -x -D cn=sgiamadmin,dc=seagate,dc=com -f {const.CORTXUSER_ACCOUNT_LDIF}')
+        Log.info("Openldap configuration completed for Cortx users.")
+
+    def _setup_ldap_permissions(self):
+        """
+        Setup necessary access permissions
+        """
+        #ToDo: Read base_dn from Conf Store once it is available
+        bind_base_dn = 'cn=sgiamadmin,dc=seagate,dc=com'
+        dn = 'olcDatabase={2}mdb,cn=config'
+        self._modify_ldap_attribute(dn, 'olcAccess', '{1}to dn.sub="dc=csm,dc=seagate,dc=com" by dn.base="'+bind_base_dn+'" read by self')
+        self._modify_ldap_attribute(dn, 'olcAccess', '{1}to dn.sub="ou=accounts,dc=csm,dc=seagate,dc=com" by dn.base="'+bind_base_dn+'" write by self')
+
+    def _run_ldap_cmd(self, cmd):
+        """
+        Run command and throw error if cmd failed
+        """
+        try:
+            _err = ""
+            Log.info(f"Executing cmd: {cmd}")
+            _proc = SimpleProcess(cmd)
```

Comment:
Use `Setup._run_cmd(<cmd>)`. instead of `_proc = SimpleProcess(cmd)`

Reply:
@udayan-y21 , Implementation is different. Here I am not raising exception for _rc =68, 80. please refer to comment in the code.